recipe/scripts/activate.bat

```diff
@@ -2,17 +2,17 @@
 @REM so other GDAL installs don't pollute the environment
 
 @if defined GDAL_DATA (
-    set "_CONDA_SET_GDAL_DATA=%GDAL_DATA%"
+    set _CONDA_SET_GDAL_DATA="%GDAL_DATA%"
 )
-@set "GDAL_DATA=%CONDA_PREFIX%\Library\share\gdal"
+@set GDAL_DATA=%CONDA_PREFIX:"=\"%\Library\share\gdal"
 
 @if defined GDAL_DRIVER_PATH (
-    set "_CONDA_SET_GDAL_DRIVER_PATH=%GDAL_DRIVER_PATH%"
+    set _CONDA_SET_GDAL_DRIVER_PATH="%GDAL_DRIVER_PATH%"
 )
 
 @REM Support plugins if the plugin directory exists
 @REM i.e if it has been manually created by the user
-@set "GDAL_DRIVER_PATH=%CONDA_PREFIX%\Library\lib\gdalplugins"
-@if not exist %GDAL_DRIVER_PATH% (
```

Comment:
I'm no expert but just checking https://ss64.com/nt/set.html (see the section "Values with Spaces - using Double Quotes") it looks like the quotes will be stored unless you quote the whole thing. ie:
```
set _CONDA_SET_GDAL_DATA="%GDAL_DATA%"
```
Will store the quoted value of GDAL_DATA, but:
```
set "_CONDA_SET_GDAL_DATA=%GDAL_DATA%"
```
(the original) will store the value of GDAL_DATA without quotes (but includes spaces). 
I'm happy to accept there is a problem with `activate.bat`, I'm just thinking that the problem you are seeing maybe elsewhere. Storing quotes in the env vars may break other things...

We should probably fix up `activate.sh` also so it quotes the paths so will work with spaces (less likely on Unix, but still possible)...

Reply:
For the record, I'm no expert either.  I just stumbled across this issue when collaborating with someone that had a space in their home directory.  Thanks for the link!

I think the issue is that in the cases where there is a space somewhere in the path of `%CONDA_PREFIX%`, you're already going to have quotes in the variable, wrapping the whole expression in double quotes results in the end_quote being detected at the first quote inside the variable expression.

For your example, I'm not sure there is an issue with `%GDAL_DATA%` specifically; I thought to modify all the variable assignments that may represent paths.

I can update a comment with this PR with echo-ing out the contents of those variables if there are spaces

Reply:
> I think the issue is that in the cases where there is a space somewhere in the path of %CONDA_PREFIX%, you're already going to have quotes in the variable

But there shouldn't be quotes in the variable... If there are quotes in the env var, it may work in .bat files but not if you access them from `os.environ` etc in Python (because Python won't process the quotes). I'm wondering if there is a bug in `conda` that is causing this to fail rather than `activate.bat`... I'll see if I can find some time to investigate.

Reply:
I managed to get onto a Windows PC to check this. I think the only problem is with the line:
```
@if not exist %GDAL_DRIVER_PATH% (
```
It should read:
```
@if not exist "%GDAL_DRIVER_PATH%" (
```
The rest of `activate.bat` should stay the same. I think your PR was just inserting quotes that are unnecessary for anything other than working around the incorrect syntax for this line. Can you confirm this?

I note that the actual activation was working (gdal commands seemed to work for me), just the check for a `gdalplugins` directory wasn't quite right when there are spaces....

Reply:
I just tested the change to gdal-activate.bat you proposed and can confirm that your suggested change does make that warning go away.

I don't know if this is an issue, but checking for the presence of  %GDAL_DRIVER_PATH% doesn't seem to show (the following is console output with the change you suggested)

```
(base) C:\Users\ogi\source\repos>conda activate "C:\Users\ogi\with space"

C:\Users\ogi\source\repos>rem proj-data is installed because its license was copied over

(C:\Users\ogi\with space) C:\Users\ogi\source\repos>python
Python 3.10.5 | packaged by conda-forge | (main, Jun 14 2022, 06:57:50) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.getenv("GDAL_DRIVER_PATH")
>>> exit()

(C:\Users\ogi\with space) C:\Users\ogi\source\repos>echo %GDAL_DRIVER_PATH%
%GDAL_DRIVER_PATH%
```

If we check for the presence of some other variables...

```
(C:\Users\ogi\with space) C:\Users\ogi\source\repos>echo %GDAL_DATA%
C:\Users\ogi\with space\Library\share\gdal

(C:\Users\ogi\with space) C:\Users\ogi\source\repos>echo %CONDA_PREFIX%
C:\Users\ogi\with space
```

Perhaps this is the intended behavior that it checks the `Library\lib` directory and not `Library\bin`?

```
@set "GDAL_DRIVER_PATH=%CONDA_PREFIX%\Library\lib\gdalplugins"
```

![image](https://user-images.githubusercontent.com/646398/177372709-9489bf51-dfa2-45a3-971f-721d835461d6.png)

Reply:
You are indeed correct - the `gdalplugins` directory is under `bin` not `lib`. Do you mind fixing this up also?

I think this is new - ie gdal didn't previously create a `gdalplugins` dir on install, we just guessed that under `lib` would be a sensible place for a user to put their own drivers. If `gdal` have made a call to do it under `bin` we should follow...

Reply:
> You are indeed correct - the `gdalplugins` directory is under `bin` not `lib`. Do you mind fixing this up also?

Happy to do so!

Reply:
Follow up question, should it point to the `bin` directory on linux as well (meaning should I update those activate scripts as well?)

Reply:
> Follow up question, should it point to the bin directory on linux as well

I just checked and it is under `lib` for Linux so those activate scripts are ok. If you feel like updating the `.sh` files so they also work with spaces also that would be great! I can see that in some situations (eg bash users on Windows) it would be nice to be able to cope with spaces....

Reply:
> I just checked and it is under `lib` for Linux so those activate scripts are ok. If you feel like updating the `.sh` files so they also work with spaces also that would be great! I can see that in some situations (eg bash users on Windows) it would be nice to be able to cope with spaces....

Least I can do 👍🏻 

Might take me a few days as I just had to give up my POSIX machine up for servicing; I'll try and test with WSL or on a Linux VM.